### PR TITLE
Bug fixes, new synchronization, new design, code clean-up

### PR DESCRIPTION
**Pull request by ThomasOliverKimble**
**Description**:

Bug fixes for beta.webots.cloud:
* functioning pagination
* overflowing list when adding multiple simulations

New synchronization:
* simulations deleted with github and yaml errors
* correct simulation type on upload and sync
* yaml errors with link to user guide

New design:
* matching blue with Vscode
* new page tabs
* improved hover on buttons and modals

Code clean-up:
* Style mods for class and id names (using '-' instead of capital letters)
* Other style mods (indentation, comments, etc.)

**Screenshots**:
list
modal

### CODE REVIEW

File: httpdocs/js/webots-cloud.js [hunk 1]
```diff
@@ -164,28 +178,24 @@ document.addEventListener('DOMContentLoaded', function() {
...
-        const url = data.url.substring(0, data.url.lastIndexOf("/worlds/"));
-        competitors = `<a class="has-text-dark" href="${url}/competitors.txt" target="_blank">${data.competitors}</a>`;
-      }
+      let icon = 'question';
+      if (data.type === 'demo')
+        icon = 'chalkboard-teacher';
+      else if (data.type === 'benchmark')
+        icon = 'award';
+      else if (data.type === 'competition')
+        icon = 'trophy';
```
Comment (omichel) [inline httpdocs/js/webots-cloud.js]: Double initialization:
```suggestion
      let icon;
      if (data.type === 'demo')
        icon = 'chalkboard-teacher';
      else if (data.type === 'benchmark')
        icon = 'award';
      else if (data.type === 'competition')
        icon = 'trophy';
      else
        icon = 'question';
```

File: httpdocs/js/webots-cloud.js [hunk 2]
```diff
@@ -371,15 +396,32 @@ document.addEventListener('DOMContentLoaded', function() {
...
           const parent = old.parentNode;
           if (data.error) {
-            ModalDialog.run('Project creation error', data.error);
-            parent.removeChild(old);
+            let errorMsg = data.error;
+            if (errorMsg.startsWith('YAML file error:')) {
+              errorMsg = errorMsg + 
+                `<div class="help">More information at: 
+                  <a target="_blank" href="https://github.com/cyberbotics/webots-cloud/blob/beta/README.md#webotsyaml">
+                    github.com/cyberbotics/webots-cloud/webotsyaml
```
Comment (omichel) [inline httpdocs/js/webots-cloud.js]: That link seems misleading, isn't it?

File: httpdocs/js/webots-cloud.js [hunk 3]
```diff
@@ -579,46 +565,114 @@ document.addEventListener('DOMContentLoaded', function() {
...
-            if (data.error)
-              modal.error(data.error);
-            else {
+            if (data.error) {
+              let errorMsg = data.error;
+              if (errorMsg.startsWith('YAML file error:')) {
+                errorMsg = errorMsg + 
+                  `<div class="help">More information at: 
+                    <a target="_blank" href="https://github.com/cyberbotics/webots-cloud/blob/beta/README.md#webotsyaml">
+                      github.com/cyberbotics/webots-cloud/webotsyaml
```
Comment (omichel) [inline httpdocs/js/webots-cloud.js]: Misleading link?

**Review by omichel — CHANGES_REQUESTED** [review]

**Comment by omichel** [comment]
Also, I believe we should not (yet) change the order of the tabs and not yet add a "Competition" tab, as I am not sure whether competitions should be listed under the "Simulation" tab or under their own tab.

File: httpdocs/js/webots-cloud.js (hunk 2 above)
Comment (ThomasOliverKimble) [reply httpdocs/js/webots-cloud.js]: I have changed it to matching links (they do not exist yet, but will exist once this PR is merged)

File: httpdocs/js/webots-cloud.js (hunk 3 above)
Comment (ThomasOliverKimble) [reply httpdocs/js/webots-cloud.js]: Same as above

**Comment by ThomasOliverKimble** [comment]
> Also, I believe we should not (yet) change the order of the tabs and not yet add a "Competition" tab, as I am not sure whether competitions should be listed under the "Simulation" tab or under their own tab.

Done

**Review by omichel — APPROVED** [review]
Very good.
Thank you.

**Pull request merged**